things/lift/modbus/js/main.js

```diff
@@ -0,0 +1,179 @@
+const { ServerTCP } = require("modbus-serial")
+const fs = require('fs')
+const path = require('path')
+const { JsonPlaceholderReplacer } = require('json-placeholder-replacer')
+const { parseArgs } = require('node:util')
+const express = require('express')
+require('dotenv').config()
+
+const app = express()
+
+const thingName = "modbus-lift"
+const hostname = "0.0.0.0"
+let portNumber = "8502"
+const thingUnitID = 1
+
+const { values: { port } } = parseArgs({
+    options: {
+        port: {
+            type: 'string',
+            short: 'p'
+        }
+    }
+})
+  
+if (port && !isNaN(parseInt(port))) {
+    portNumber = parseInt(port)
+}
+
+const tmPath = process.env.TM_PATH
+
+if (process.platform === 'win32') {
+    tmPath.split(path.sep).join(path.win32.sep)
+}
+
+const thingModel = JSON.parse(fs.readFileSync(path.join(__dirname, tmPath)))
+
+const placeholderReplacer = new JsonPlaceholderReplacer()
+placeholderReplacer.addVariableMap({
+    PROTOCOL: 'modbus+tcp',
+    THING_NAME: thingName,
+    HOSTNAME: hostname,
+    PORT_NUMBER: portNumber
+})
+const thingDescription = placeholderReplacer.replace(thingModel)
+thingDescription['@type'] = 'Thing'
+
+const coils = new Array(9999)
+const discreteInputs = new Array(9999)
+const inputRegisters = new Array(9999)
+const holdingRegisters = new Array(9999)
+
+const lightSwitchForms = [{
+    "href": `${thingUnitID}?address=1&quantity=1`,
```

Comment:
unitID should be at the base

Reply:
Fixed at [1e7bc99](https://github.com/eclipse-thingweb/test-things/pull/9/commits/1e7bc99a0e815868f9254edfff03ef9562635cd2)